### PR TITLE
feat(eslint-plugin): add eslint rule no-useless-constructor

### PR DESCRIPTION
**Pull request by armano2**
This PR adds wrapper around eslint `no-useless-constructor` to fix issues from #15 and #48

fixes: #15

### CODE REVIEW

File: packages/eslint-plugin/docs/rules/no-useless-constructor.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+class A extends B {
+  constructor(...args) {
+    super(...args);
+  }
+}
+```
+
+Examples of **correct** code for this rule:
```
Comment (bradzacher) [inline packages/eslint-plugin/docs/rules/no-useless-constructor.md]: would suggest adding an example using parameter properties considering that's the point of this overload!

File: packages/eslint-plugin/lib/rules/no-useless-constructor.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+const baseRule = require('eslint/lib/rules/no-useless-constructor');
+const util = require('../util');
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = Object.assign({}, baseRule, {
+  meta: {
```
Comment (bradzacher) [inline packages/eslint-plugin/lib/rules/no-useless-constructor.js]: I would class this as recommended: error

Comment (armano2) [reply packages/eslint-plugin/lib/rules/no-useless-constructor.js]: we should not enable rules without major release

File: packages/eslint-plugin/tests/lib/rules/no-useless-constructor.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    'class A extends B { constructor(foo, bar) { super(bar); } }',
+    // https://github.com/typescript-eslint/typescript-eslint/issues/15
+    'declare class Foo { constructor(); }',
+    'class Foo { constructor(); }',
+    'abstract class Foo { constructor(); }',
+    'abstract class Foo { abstract constructor(); }',
+    // https://github.com/typescript-eslint/typescript-eslint/issues/48
+    'class Foo { constructor(private name: string) {} }',
+    'class Foo { constructor(public name: string) {} }',
+    'class Foo { constructor(protected name: string) {} }'
```
Comment (mysticatea) [inline packages/eslint-plugin/tests/lib/rules/no-useless-constructor.js]: I'd like to see a test for the constructor with visibility.

```ts
class Foo { private constructor() {} } // valid!
```

Comment (armano2) [reply packages/eslint-plugin/tests/lib/rules/no-useless-constructor.js]: i didn't think about accessibility, i will add it

File: packages/eslint-plugin/lib/rules/no-useless-constructor.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+    return {
+      MethodDefinition(node) {
+        if (
+          !node.value ||
+          node.value.type !== 'FunctionExpression' ||
+          node.accessibility
```
Comment (bradzacher) [inline packages/eslint-plugin/lib/rules/no-useless-constructor.js]: would it be good to check for not public?

i.e. this use constructor is still useless:
```TS
class X {
  public constructor() {}
}
```

Comment (armano2) [reply packages/eslint-plugin/lib/rules/no-useless-constructor.js]: public one should be useless only when there is no super class?

Comment (bradzacher) [reply packages/eslint-plugin/lib/rules/no-useless-constructor.js]: you are correct [repl](https://www.typescriptlang.org/play/index.html#src=class%20Parent%20%7B%0A%20%20%20%20protected%20constructor()%20%7B%20%7D%0A%7D%0A%0Aclass%20Child%20extends%20Parent%20%7B%20%7D%0A%0Aconst%20x%20%3D%20new%20Child()%3B%0A)

the public constructor def is only useless if there is no super class.

technically the public case is *not* useless if:
- there is a super class
- the super class defines a constructor
- the constructor has a different signature

But that's not checkable from within eslint without full type info

Comment (armano2) [reply packages/eslint-plugin/lib/rules/no-useless-constructor.js]: ok, than i'm just going to check if there is super class and add test cases for that

**Comment by tsofist** [comment]
I still have a problem with `d.ts` file :(
```ts
declare class MySuperClass {
    constructor(argument1: number);
}
```

**Error**
```
TypeError: Cannot read property 'body' of null
Occurred while linting test.d.ts:2
    at checkForConstructor (node_modules\eslint\lib\rules\no-useless-constructor.js:169:42)
```

**Comment by bradzacher** [comment]
please don't necro old PRs.
Open an issue instead.

**Pull request merged**